doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,410 @@
+FORMAT: 1.0
+
+# REST service API for sharing configuration snippets
+
+This is a description of the API provided by the REST service that can be used to share, search and download configuration snippets.
+
+
+# API Description [/]
+
+## List API methods [GET]
+
+Returns an overview of available API methods with required parameters and a description.
+
++ Response 200 (text/html; charset=utf-8)
+
+
+
+# Authentication endpoint [/auth]
+
+## Authenticate [POST]
+
+The API will try to authenticate the user by the submitted credentials. In case of success a session token (JWT) will be returned, which can be used for further requests to the API.
+
++ Request (application/x-www-form-urlencoded)
+    + Attributes
+        + username: the_username (string, required)
+        + password: aSecretPassword (string, required)
```

Comment:
> The proposed method does not offer protection during establishment of passwords

In this case I can only think of using HTTPS instead.

> I'm not sure if challenge/response systems are ideal to be used with HTTP anyway...?

It is certainly not ideal. HTTPS would be better. But challenge/response offers better protection of the password itself than plaintext or hashing. It does not make the authentication process more secure but many users tend to re-use their passwords across multiple service. So every countermeasure against password sniffing is important.